rs-commerce/src/components/login-page-form/login-page-form.tsx

```diff
@@ -0,0 +1,180 @@
+import {
+  EMAIL_FORM_TIPS,
+  LOGIN_PAGE_TEXT,
+  PASSWORD_FORM_TIPS,
+  ROUTE_PATH,
+} from 'constants/constants';
+import { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+import {
+  checkEmailDomainIsValid,
+  checkEmailHasAmpSymbol,
+  checkEmailHasDomain,
+  checkEmailHasLocalPart,
+  checkEmailLocalPartIsValid,
+  checkEmailValid,
+  checkIsValueEmpty,
+  checkIsValueHasDigit,
+  checkIsValueHasLowercase,
+  checkIsValueHasOnlyLatin,
+  checkIsValueHasSpecial,
+  checkIsValueHasUppercase,
+  checkIsValueHasWhitespaces,
+  checkPasswordLength,
+} from 'utils/check-utils';
+import style from './style.module.css';
+
+function LoginPageForm(): JSX.Element {
+  const [emailValue, setEmailValue] = useState('');
+  const [emailTouched, setEmailTouched] = useState(false);
+  const [isEmailValid, setIsEmailValid] = useState(false);
+  const [emailTips, setEmailTips] = useState('');
+  const [passwordValue, setPasswordValue] = useState('');
+  const [passwordTouched, setPasswordTouched] = useState(false);
+  const [isPasswordValid, setIsPasswordValid] = useState(false);
+  const [passwordTips, setPasswordTips] = useState('');
+  const [isOpenEye, setIsOpenEye] = useState(false);
+  const [isCanSubmit, setIsCanSubmit] = useState(false);
+
+  const checkIsEmailValid = (): void => {
+    if (!emailTouched) return;
+
+    if (checkIsValueEmpty(emailValue)) {
+      setEmailTips(EMAIL_FORM_TIPS.tipsEmailEmpty);
+    } else if (checkIsValueHasWhitespaces(emailValue)) {
+      setEmailTips(EMAIL_FORM_TIPS.tipsEmailWhitespaces);
+    } else if (!checkEmailHasAmpSymbol(emailValue)) {
+      setEmailTips(EMAIL_FORM_TIPS.tipsEmailAmpSymbol);
+    } else if (checkEmailHasLocalPart(emailValue)) {
+      setEmailTips(EMAIL_FORM_TIPS.tipsEmailLocalPart);
+    } else if (checkEmailLocalPartIsValid(emailValue)) {
+      setEmailTips(EMAIL_FORM_TIPS.tipsEmailLocalPartValid);
+    } else if (checkEmailHasDomain(emailValue)) {
+      setEmailTips(EMAIL_FORM_TIPS.tipsEmailDomain);
+    } else if (checkEmailDomainIsValid(emailValue)) {
+      setEmailTips(EMAIL_FORM_TIPS.tipsEmailDomainValid);
+    } else if (!checkEmailValid(emailValue)) {
+      setEmailTips(EMAIL_FORM_TIPS.tipsEmailFormat);
+    } else {
+      setEmailTips('');
+      setIsEmailValid(true);
+    }
+
+    setIsEmailValid(false);
```

Comment:
I think ` return;` in the last else is required because in other case  `setIsEmailValid(false);`  will be invoked always and after `  setIsEmailValid(true);` true.
Does your decision work properly?

Reply:
I did it